tests/unit/linalg/test_amgfsolver.cpp

```diff
@@ -0,0 +1,239 @@
+// CONTRIBUTING.md for details.
+
+#include "unit_tests.hpp"
+#include "mfem.hpp"
+#include "../../config/config.hpp"
+
+namespace mfem
+{
+#ifdef MFEM_USE_MPI
+
+class DenseMatrixSolver : public Solver
+{
+private:
+   DenseMatrix A;
+   LUFactors LU;
+   int * ipiv = nullptr;
+public:
+   DenseMatrixSolver() : Solver() {}
+
+   void SetOperator(const Operator &op) override
+   {
+      auto Oph = const_cast<HypreParMatrix *>(dynamic_cast<const HypreParMatrix *>
+                                              (&op));
+      MFEM_VERIFY(Oph, "Not a compatible matrix type");
+      SparseMatrix Sp;
+      Oph->MergeDiagAndOffd(Sp);
+      Sp.ToDenseMatrix(A);
+      delete [] ipiv;
+      ipiv = new int[A.Height()];
+      LU.data = A.Data();
+      LU.ipiv = ipiv;
+      LU.Factor(A.Height());
+   }
+
+   void Mult(const Vector &x, Vector &y) const override
+   {
+      y = x;
+      LU.Solve(A.Height(), 1, y.GetData());
+   }
+
+   ~DenseMatrixSolver() { delete [] ipiv; }
+};
+
+HypreParMatrix * GetProlongationMatrix(const ParFiniteElementSpace* pfes,
+                                       int element_attribute)
+{
+   Array<int> dofs;
+   Array<int> tdofs;
+   for (int i = 0; i < pfes->GetNE(); i++)
+   {
+      if (pfes->GetAttribute(i) == element_attribute)
+      {
+         pfes->GetElementVDofs(i, dofs);
+         tdofs.Append(dofs);
+      }
+   }
+   tdofs.Sort();
+   tdofs.Unique();
+
+   HYPRE_BigInt h = tdofs.Size();
+   SparseMatrix St(h,pfes->GlobalTrueVSize());
+
+   for (int i = 0; i<h; i++)
+   {
+      int col = tdofs[i];
+      St.Set(i,col,1.0);
+   }
+   St.Finalize();
+   HYPRE_BigInt rows[2];
+   HYPRE_BigInt cols[2];
+   int nrows = St.Height();
+
+   rows[0] = 0; rows[1] = nrows;
+   for (int i = 0; i < 2; i++)
+   {
+      cols[i] = pfes->GetTrueDofOffsets()[i];
+   }
+   HYPRE_BigInt glob_nrows = nrows;
+   HYPRE_BigInt glob_ncols = pfes->GlobalTrueVSize();
+
+   HYPRE_BigInt * J;
+#ifndef HYPRE_BIGINT
+   J = St.GetJ();
+#else
+   J = new HYPRE_BigInt[St.NumNonZeroElems()];
+   for (int i = 0; i < St.NumNonZeroElems(); i++)
+   {
+      J[i] = St.GetJ()[i];
+   }
+#endif
+   HypreParMatrix * Pt = new HypreParMatrix(pfes->GetComm(), nrows, glob_nrows,
+                                            glob_ncols, St.GetI(), J,
+                                            St.GetData(), rows,cols);
+
+   HypreParMatrix * P = Pt->Transpose();
+   delete Pt;
+#ifdef HYPRE_BIGINT
+   delete [] J;
+#endif
+   return P;
+}
+
+
+TEST_CASE("FilteredSolver and AMGFSolver", "[Parallel]")
+{
+   int myid = Mpi::WorldRank();
+
+   // Note: This test is restricted to a single processor for convenience,
+   // allowing the use of a serial dense direct solver on the filtered subspace
+   // and avoiding any dependency on external parallel sparse direct solvers.
+   // In general, both AMGFSolver and FilteredSolver are designed to work in parallel.
+   if (myid == 0)
```

Comment:
```suggestion
   if (Mpi::Root())
```

And can then delete the variable `myid`.